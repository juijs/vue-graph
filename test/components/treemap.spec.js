import { mount } from '@vue/test-utils'
import GraphTreeMap from '@/components/treemap'

describe('treemap.js', () => {
    it('snapshot', () => {
        const wrapper = mount(GraphTreeMap, {
            props: {
                width: 800,
                height: 800,
                textAlign: 'right',
                textVerticalAlign: 'bottom',
                colors: [ '#EC2500', '#ECE100', '#EC9800', '#9EDE00' ],
                values: [
                    [ '0', 'Apples', -1 ],
                    [ '0.0', 'Anne', 5 ],
                    [ '0.1', 'Rick', 3 ],
                    [ '0.2', 'Peter', 4 ],
                    [ '1', 'Bananas', -1 ],
                    [ '1.0', 'Anne', 4 ],
                    [ '1.1', 'Rick', 10 ],
                    [ '1.2', 'Peter', 1 ],
                    [ '2', 'Oranges', -1 ],
                    [ '2.0', 'Anne', 1 ],
                    [ '2.1', 'Rick', 3 ],
                    [ '2.2', 'Peter', 3 ],
                    [ '3', 'Susanne', 2 ],
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});