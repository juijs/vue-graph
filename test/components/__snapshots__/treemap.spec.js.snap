// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`treemap.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="800"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="800"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="700"
            width="700"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="700"
            width="700"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="700"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,50)"
        width="700"
        x="0"
        y="0"
      />
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,50)"
      >
        <rect
          fill="transparent"
          height="700"
          stroke="transparent"
          width="700"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-treemap"
        transform="translate(50,50)"
      >
        <rect
          fill="#9EDE00"
          height="155.55555555555554"
          stroke="#333"
          stroke-width="0.5"
          width="175"
          x="525"
          y="544.4444444444445"
        />
        <rect
          fill="#EC9800"
          height="233.33333333333331"
          stroke="#333"
          stroke-width="0.5"
          width="175"
          x="525"
          y="311.11111111111114"
        />
        <rect
          fill="#EC9800"
          height="233.33333333333337"
          stroke="#333"
          stroke-width="0.5"
          width="175"
          x="525"
          y="77.77777777777779"
        />
        <rect
          fill="#EC9800"
          height="77.77777777777779"
          stroke="#333"
          stroke-width="0.5"
          width="175"
          x="525"
          y="0"
        />
        <rect
          fill="#ECE100"
          height="35.353535353535335"
          stroke="#333"
          stroke-width="0.5"
          width="385.0000000000009"
          x="140"
          y="664.6464646464647"
        />
        <rect
          fill="#ECE100"
          height="353.5353535353535"
          stroke="#333"
          stroke-width="0.5"
          width="385"
          x="140"
          y="311.11111111111114"
        />
        <rect
          fill="#ECE100"
          height="388.88888888888886"
          stroke="#333"
          stroke-width="0.5"
          width="140"
          x="0"
          y="311.11111111111114"
        />
        <rect
          fill="#EC2500"
          height="177.7777777777778"
          stroke="#333"
          stroke-width="0.5"
          width="306.25"
          x="218.75"
          y="133.33333333333334"
        />
        <rect
          fill="#EC2500"
          height="133.33333333333334"
          stroke="#333"
          stroke-width="0.5"
          width="306.25"
          x="218.75"
          y="0"
        />
        <rect
          fill="#EC2500"
          height="311.11111111111114"
          stroke="#333"
          stroke-width="0.5"
          width="218.75"
          x="0"
          y="0"
        />
        <text
          fill="#333"
          font-size="12"
          font-weight="bold"
          text-anchor="start"
          x="3"
          y="12"
        >
          Apples
        </text>
        <text
          fill="#333"
          font-size="12"
          font-weight="bold"
          text-anchor="start"
          x="3"
          y="323.11111111111114"
        >
          Bananas
        </text>
        <text
          fill="#333"
          font-size="12"
          font-weight="bold"
          text-anchor="start"
          x="528"
          y="12"
        >
          Oranges
        </text>
        <text
          fill="#333"
          font-size="12"
          font-weight="bold"
          text-anchor="start"
          x="528"
          y="556.4444444444445"
        >
          Susanne
        </text>
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
